App fails to start when connection to OIDC server times out

### Describe the bug

When the connection to the OIDC server times out on startup the app crashes with the following error:

```
2024-06-28 12:07:35,007 WARN  [io.qua.oid.run.OidcRecorder] (vert.x-eventloop-thread-1) OIDC server is not available at the 'http://idp--test:8082/auth/realms/master' URL. Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example: 'https://localhost:8180/auth/realms/quarkus'
2024-06-28 12:07:35,026 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: io.smallrye.mutiny.TimeoutException
	at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:64)
	at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
	at io.quarkus.oidc.runtime.OidcRecorder.createStaticTenantContext(OidcRecorder.java:173)
	at io.quarkus.oidc.runtime.OidcRecorder.setup(OidcRecorder.java:91)
	at io.quarkus.deployment.steps.OidcBuildStep$setup1958257566.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.OidcBuildStep$setup1958257566.deploy(Unknown Source)
	... 11 more
```

### Expected behavior

The appliaction starts and attempts to connect during the first request. This is already the behaviour for any other connection error.

### Actual behavior

_No response_

### How to Reproduce?

- configure an odic server which does not respond within 10 seconds
- start the quarkus app

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java
@@ -55,6 +55,7 @@ import io.quarkus.tls.TlsConfiguration;
 import io.quarkus.tls.TlsConfigurationRegistry;
 import io.smallrye.jwt.algorithm.KeyEncryptionAlgorithm;
 import io.smallrye.jwt.util.KeyUtils;
+import io.smallrye.mutiny.TimeoutException;
 import io.smallrye.mutiny.Uni;
 import io.vertx.core.Vertx;
 import io.vertx.core.json.JsonObject;
@@ -167,29 +168,37 @@ public class OidcRecorder {
 
         Uni<TenantConfigContext> uniContext = createTenantContext(vertx, oidcConfig, checkNamedTenants, tenantId,
                 defaultTlsConfiguration);
-        return uniContext.onFailure()
-                .recoverWithItem(new Function<Throwable, TenantConfigContext>() {
-                    @Override
-                    public TenantConfigContext apply(Throwable t) {
-                        if (t instanceof OIDCException) {
-                            LOG.warnf("Tenant '%s': '%s'."
-                                    + " OIDC server is not available yet, an attempt to connect will be made during the first request."
-                                    + " Access to resources protected by this tenant may fail"
-                                    + " if OIDC server will not become available",
-                                    tenantId, t.getMessage());
-                            return new TenantConfigContext(null, oidcConfig, false);
+        try {
+            return uniContext.onFailure()
+                    .recoverWithItem(new Function<Throwable, TenantConfigContext>() {
+                        @Override
+                        public TenantConfigContext apply(Throwable t) {
+                            if (t instanceof OIDCException) {
+                                LOG.warnf("Tenant '%s': '%s'."
+                                        + " OIDC server is not available yet, an attempt to connect will be made during the first request."
+                                        + " Access to resources protected by this tenant may fail"
+                                        + " if OIDC server will not become available",
+                                        tenantId, t.getMessage());
+                                return new TenantConfigContext(null, oidcConfig, false);
+                            }
+                            logTenantConfigContextFailure(t, tenantId);
+                            if (t instanceof ConfigurationException
+                                    && !oidcConfig.authServerUrl.isPresent()
+                                    && LaunchMode.DEVELOPMENT == LaunchMode.current()) {
+                                // Let it start if it is a DEV mode and auth-server-url has not been configured yet
+                                return new TenantConfigContext(null, oidcConfig, false);
+                            }
+                            // fail in all other cases
+                            throw new OIDCException(t);
                         }
-                        logTenantConfigContextFailure(t, tenantId);
-                        if (t instanceof ConfigurationException
-                                && !oidcConfig.authServerUrl.isPresent() && LaunchMode.DEVELOPMENT == LaunchMode.current()) {
-                            // Let it start if it is a DEV mode and auth-server-url has not been configured yet
-                            return new TenantConfigContext(null, oidcConfig, false);
-                        }
-                        // fail in all other cases
-                        throw new OIDCException(t);
-                    }
-                })
-                .await().atMost(oidcConfig.getConnectionTimeout());
+                    })
+                    .await().atMost(oidcConfig.getConnectionTimeout());
+        } catch (TimeoutException t2) {
+            LOG.warnf("Tenant '%s': OIDC server is not available after a %d seconds timeout, an attempt to connect will be made"
+                    + " during the first request. Access to resources protected by this tenant may fail if OIDC server"
+                    + " will not become available", tenantId, oidcConfig.getConnectionTimeout().getSeconds());
+            return new TenantConfigContext(null, oidcConfig, false);
+        }
     }
 
     private static Throwable logTenantConfigContextFailure(Throwable t, String tenantId) {

